### PR TITLE
Fallback to Clojars for finding artifact version

### PR DESCRIPTION
**Pull request by yannvanhalewyn**
I had opened issue #267. This is the PR to solve the issue. This will use the Clojars API to fetch artifact versions if there are none to be found in Maven. The goal of this is to allow `cljr-update-project-dependency(ies)` to work with Clojars-only libraries, such as clj-http, clj-time and aero.

### CODE REVIEW

File: src/refactor_nrepl/artifacts.clj [hunk 1]
```diff
@@ -101,6 +101,14 @@
                    (map #(vector (str group-id "/" %) nil))))
             group-ids)))
 
+(defn get-clojars-versions!
+  "Fetches all the versions of particular artifact from Clojars."
+  [for-artifact]
```
Comment (bbatsov) [inline src/refactor_nrepl/artifacts.clj]: I'd name this just `artifact`.

File: src/refactor_nrepl/artifacts.clj [hunk 2]
```diff
@@ -125,7 +133,8 @@
 (defn artifact-versions
   [{:keys [artifact]}]
   (->> (or (get @artifacts artifact)
-           (get-mvn-versions! artifact))
+           (seq (get-mvn-versions! artifact))
```
Comment (bbatsov) [inline src/refactor_nrepl/artifacts.clj]: Might be a good idea to explain the behaviour of this function in the docstring.

File: src/refactor_nrepl/artifacts.clj (hunk 1 above)
Comment (yannvanhalewyn) [reply src/refactor_nrepl/artifacts.clj]: Me too, I followed the convention of the other function. Will update!

File: src/refactor_nrepl/artifacts.clj (hunk 2 above)
Comment (yannvanhalewyn) [reply src/refactor_nrepl/artifacts.clj]: 👍

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**